dev/deps/spark-deps-hadoop-2.6

```diff
@@ -21,8 +21,6 @@ automaton-1.11-8.jar
 avro-1.7.7.jar
 avro-ipc-1.7.7.jar
 avro-mapred-1.7.7-hadoop2.jar
-base64-2.3.8.jar
-bcprov-jdk15on-1.58.jar
```

Comment:
Hm, so bouncy castle is also removed? Maybe I miss what bcprov is, but I think it's the "bouncycastle provider". If so, the license in licenses-binary/ and in LICENSE-binary should be removed. And there's an entry separately in the parent POM that brings this in on account of jets3t, it says. That can go?

Reply:
if there's nothing else for bouncy castle then it's cleaner for classpaths if it goes, as for imports.

looking at Hadoop, it it pulls in bcprov-jdk16, but only for testing. and with out any explicit calls on bouncy castle APIs that I can see. Presumably its registered as a crypto

```xml
     <dependency>
       <groupId>org.bouncycastle</groupId>
       <artifactId>bcprov-jdk16</artifactId>
       <version>1.46</version>
       <scope>test</scope>
     </dependency>
```

If it is going, I should update this PR & JIRA name to make clear it is gone

Reply:
Sounds like it's already been removed, so the other 'remnants' should go.

Reply:
@steveloughran I'll merge this if you can remove the bouncy castle license info and take one more pass at comments here.